target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c

```diff
@@ -1342,6 +1366,15 @@ static int ag71xx_probe(struct platform_device *pdev)
 	if (!np)
 		return -ENODEV;
 
+	err = ag71xx_setup_gmac(np);
+	if (err)
+		return err;
+
+	if (of_device_is_compatible(np, "qca,ar9340-eth") && (!ag71xx_gmac_configured)) {
```

Comment:
I didn't realize that this problem exists on other SoCs and I applied this workaround only for ar934x.
@lucize You should add a compatible string to use this workaround.

Reply:
yes, I meant to ask you about this, I'll try

Reply:
I added something like 

`if ((of_device_is_compatible(np, "qca,ar9340-eth") || of_device_is_compatible(np, "qca,qca9530-eth")) && (!ag71xx_gmac_configured)) {`

Reply:
As @lucize tested it on WR841v9, the qca953x soc´s does not need this fix.

Reply:
with

[    0.625225] libphy: Fixed MDIO Bus: probed
[    1.311983] libphy: mdio: probed
[    1.682981] mdio-bus.0: Found an AR934X built-in switch
[    1.763668] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.773753] eth0: Atheros AG71xx at 0xba000000, irq 5, mode:GMII
[    2.481983] libphy: mdio: probed
[    2.486817] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:04 [uid=004dd042, driver=Generic PHY]
[    2.497514] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode:MII

without

[    0.400919] libphy: Fixed MDIO Bus: probed
[    1.083694] libphy: mdio: probed
[    1.087067] ag71xx 19000000.eth: Could not connect to PHY device
[    1.773690] libphy: mdio: probed
[    2.114686] mdio-bus.1: Found an AR934X built-in switch
[    2.169405] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    2.179477] eth0: Atheros AG71xx at 0xba000000, irq 5, mode:GMII

Reply:
Also if you disable/remove mdio0

Reply:
if I don't have mdio0 then eth0 or eth1 will not show, one of them, I don't recall, but I tested that also

Reply:
now is like on ar71xx target, eth1 being the wan, before this patch, eth0 was the wan as port 5 on the switch, now I have only up to port4 on the switch (like ar71xx)